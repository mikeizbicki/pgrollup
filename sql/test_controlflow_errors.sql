SET client_min_messages TO WARNING;

-- if we call pg_rollup functinos on a table that doesn't exist, they should throw an informative error

select do_rollup('table_does_not_exist');

select rollup_mode('table_does_not_exist', 'trigger');

