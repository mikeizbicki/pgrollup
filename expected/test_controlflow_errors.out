SET client_min_messages TO WARNING;
-- if we call pg_rollup functinos on a table that doesn't exist, they should throw an informative error
select do_rollup('table_does_not_exist');
ERROR:  no rollups named table_does_not_exist
CONTEXT:  PL/pgSQL function do_rollup(text,text,bigint,boolean,integer) line 32 at RAISE
select rollup_mode('table_does_not_exist', 'trigger');
ERROR:  relation "table_does_not_exist" does not exist
LINE 1: select rollup_mode('table_does_not_exist', 'trigger');
                           ^
